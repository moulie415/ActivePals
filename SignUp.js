import React, { Component } from "react"
import { StyleSheet, Alert } from "react-native"
import {
  Button,
  Text,
  Input,
  Container,
  Item,
  Header,
  Left,
  Icon,
  Body,
  Title,
  Right,
  Content,
  Spinner
} from 'native-base'
import * as firebase from "firebase"
import  styles  from './styles/signUpStyles'

 export default class SignUp extends Component {

  constructor(props) {
    super(props)

    this.username = ""
    this.pass = ""
    this.confirm = ""
    this.state = {
      spinner: false
    }
  }

  componentDidMount() {

    firebase.auth().onAuthStateChanged(function(user) {
      if (user) {
        this.user = user
    // User is signed in.
  } else {
    // No user is signed in.
  }
})
  }

  render () {
    return (
    <Container style={styles.container}>
      {this.state.spinner && <Spinner />}
      <Item rounded style={styles.inputGrp}>
        <Input
        placeholder="Email"
        onChangeText={u => this.username = u}
        placeholderTextColor="#FFF"
        style={styles.input}
        autoCapitalize={'none'}
        autoCorrect={false}
        //value={this.state.username}
        keyboardType={'email-address'}
        />
        </Item>
      <Item rounded style={styles.inputGrp}>
      <Input
        placeholder="Password"
        secureTextEntry={true}
        placeholderTextColor="#FFF"
        onChangeText={p => this.pass = p}
        style={styles.input}
        />
        </Item>
      <Item rounded style={styles.inputGrp}>
      <Input
        placeholder="Confirm Password"
        secureTextEntry={true}
        placeholderTextColor="#FFF"
        onChangeText={p => this.confirm = p}
        style={styles.input}
        />
        </Item>
        <Button primary rounded
        onPress={() => {
          if (this.pass == this.confirm) {
            this.setState({spinner: true})
            this.signup(this.username, this.pass)
          }
          else {
            Alert.alert("Please try again", "Passwords do not match")
          }
        }}
        style={{alignSelf: 'center'}}
        >
        <Text>Sign up</Text>
        </Button>
    </Container>
  )
  }

    async signup(email, pass) {

      try {
        await firebase.auth()
        .createUserWithEmailAndPassword(email, pass);

        this.setState({spinner: false})
        console.log("Account created");
        Alert.alert("account created")

      } catch (error) {
        console.log(error.toString())
        this.setState({spinner: false})
        Alert.alert(error.toString())
      }

} 

} 
