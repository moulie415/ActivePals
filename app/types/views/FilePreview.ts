import Profile from '../Profile';
import {StackNavigationProp} from '@react-navigation/stack';
import {RouteProp} from '@react-navigation/native';
import {StackParamList} from '../../App';
import Post from '../Post';

type FilePreviewNavigationProp = StackNavigationProp<
  StackParamList,
  'FilePreview'
>;
type FilePreviewRouteProp = RouteProp<StackParamList, 'FilePreview'>;

export default interface FilePreviewProps {
  navigation: FilePreviewNavigationProp;
  route: FilePreviewRouteProp;
  friends: {[key: string]: Profile};
  profile: Profile;
  postStatus: (status: Post) => void;
  setPostMessage: (url: string, text: string) => void;
};
