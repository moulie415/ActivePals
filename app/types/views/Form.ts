import { NavigationStackProp } from "react-navigation-stack";

export default interface FormProps {
  navigation: NavigationStackProp;
}
