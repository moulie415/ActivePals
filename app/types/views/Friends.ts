import Profile from '../Profile';
import {NavigationStackProp} from 'react-navigation-stack';
import {ThemeType} from '@ui-kitten/components';

export default interface FriendsProps {
  profile: Profile;
  friends: {[key: string]: Profile};
  onRequest: (key: string) => void;
  getFriends: (uid: string, limit?: number, startAt?: string) => void;
  onRemove: (uid: string) => void;
  onAccept: (uid: string, friendUid: string) => void;
  navigation: NavigationStackProp;
  setModal: (show: boolean) => void;
  modalOpen: boolean;
  eva?: ThemeType;
}
