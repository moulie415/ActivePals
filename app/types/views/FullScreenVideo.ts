import {StackNavigationProp} from '@react-navigation/stack';
import {StackParamList} from '../../App';
import {RouteProp} from '@react-navigation/native';

type FullScreenVideoNavigationProp = StackNavigationProp<
  StackParamList,
  'FullScreenVideo'
>;

type FullScreenVideoRouteProp = RouteProp<StackParamList, 'FullScreenVideo'>;
export default interface FullScreenVideoProps {
  navigation: FullScreenVideoNavigationProp;
  route: FullScreenVideoRouteProp;
};
