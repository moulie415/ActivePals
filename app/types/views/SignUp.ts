export default interface SignUpProps {
  goBack: () => void;
}
