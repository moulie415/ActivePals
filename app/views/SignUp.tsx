import React, { Component } from 'react';
import { Alert, View, ImageBackground, KeyboardAvoidingView, TextInput } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import firebase from 'react-native-firebase';
import styles from '../styles/signUpStyles';
import colors from '../constants/colors';
import Header from '../components/Header/header';
import { PulseIndicator } from 'react-native-indicators';
import sStyles from '../styles/settingsStyles';
const background = require('Anyone/assets/images/Running-background.jpg');
import Button from '../components/Button';

class SignUp extends Component {
  static navigationOptions = {
    header: null,
  };

  constructor(props) {
    super(props);

    this.username = '';
    this.email = '';
    this.pass = '';
    this.confirm = '';
    this.state = {
      spinner: false,
    };
  }

  render() {
    return (
      <ImageBackground style={styles.container} source={background}>
        <Header hasBack={true} title={'Sign up'} />
        {this.state.spinner && (
          <View style={sStyles.spinner}>
            <PulseIndicator color={colors.secondary} />
          </View>
        )}
        <KeyboardAvoidingView behavior="padding" style={{ justifyContent: 'center', flex: 1 }}>
          <View style={styles.inputGrp}>
            <Icon size={25} name="ios-person" style={{ color: '#fff', marginRight: 5 }} />
            <TextInput
              placeholder="Username"
              onChangeText={u => (this.username = u)}
              placeholderTextColor="#FFF"
              style={styles.input}
              autoCapitalize={'none'}
              autoCorrect={false}
              keyboardType={'email-address'}
            />
          </View>
          <View style={styles.inputGrp}>
            <Icon size={25} name="md-mail" style={styles.icon} />
            <TextInput
              placeholder="Email"
              onChangeText={e => (this.email = e)}
              placeholderTextColor="#FFF"
              style={styles.input}
              autoCapitalize={'none'}
              autoCorrect={false}
              keyboardType={'email-address'}
            />
          </View>
          <View style={styles.inputGrp}>
            <Icon size={25} name="md-unlock" style={styles.icon} />
            <TextInput
              placeholder="Password"
              secureTextEntry={true}
              placeholderTextColor="#FFF"
              onChangeText={p => (this.pass = p)}
              style={styles.input}
            />
          </View>
          <View style={styles.inputGrp}>
            <Icon size={25} name="md-unlock" style={styles.icon} />
            <TextInput
              placeholder="Confirm Password"
              secureTextEntry={true}
              placeholderTextColor="#FFF"
              onChangeText={p => (this.confirm = p)}
              style={styles.input}
            />
          </View>
          <Button
            style={{ paddingHorizontal: 20, alignSelf: 'center' }}
            onPress={() => {
              if (this.pass === this.confirm) {
                this.setState({ spinner: true });
                if (this.username && this.username.length > 5 && !str.whiteSpaceRegex.test(this.username)) {
                  firebase
                    .database()
                    .ref('/usernames/' + this.username)
                    .once('value')
                    .then(snapshot => {
                      if (snapshot.val()) {
                        Alert.alert('Sorry', 'That username is already in use');
                        this.setState({ spinner: false });
                      } else {
                        this.signup(this.email, this.pass);
                      }
                    });
                } else {
                  Alert.alert('Sorry', 'Username must be at least 5 characters long and cannot contain any spaces');
                  this.setState({ spinner: false });
                }
              } else {
                Alert.alert('Please try again', 'Passwords do not match');
                this.setState({ spinner: false });
              }
            }}
            text="Sign up"
          />
        </KeyboardAvoidingView>
      </ImageBackground>
    );
  }

  async signup(email, pass) {
    try {
      await firebase
        .auth()
        .createUserWithEmailAndPassword(email, pass)
        .then(({ info, user }) => {
          const userData = { uid: user.uid, email: user.email, username: this.username };
          this.createUser(user.uid, userData, '');
          user
            .sendEmailVerification()
            .then(() => {
              this.props.goBack();
              Alert.alert(
                'Account created',
                'You must now verify your email using the link we sent you before you can login'
              );
              this.setState({ spinner: false });
            })
            .catch(error => {
              Alert.alert('Error', error.message);
              this.setState({ spinner: false });
            });
        })
        .catch(error => {
          console.log(error);
          Alert.alert('Error', error.message);
          this.setState({ spinner: false });
        });
    } catch (error) {
      console.log(error.toString());
      this.setState({ spinner: false });
      Alert.alert(error.toString());
    }
  }

  createUser = (uid, userData, token) => {
    firebase
      .database()
      .ref('admins')
      .child(uid)
      .once('value', snapshot => {
        const defaults = {
          uid,
          token,
          accountType: snapshot.val() ? AccountType.ADMIN : AccountType.STANDARD,
        };
        //Alert.alert("Success", "Logged in as: " + userData.email)
        firebase
          .database()
          .ref('users')
          .child(uid)
          .update({ ...userData, ...defaults });
        if (this.username) {
          firebase
            .database()
            .ref('usernames')
            .child(this.username)
            .set(uid);
        }
      });
  };
}

import { connect } from 'react-redux';
import { navigateBack } from '../actions/navigation';
import { AccountType } from '../types/Profile';
import str from '../constants/strings';
// const mapStateToProps = ({ home, settings, profile }) => ({
// })

const mapDispatchToProps = dispatch => ({
  goBack: () => dispatch(navigateBack()),
});

export default connect(null, mapDispatchToProps)(SignUp);
