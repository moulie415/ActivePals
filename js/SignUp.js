import React, { Component } from "react"
import {
  Alert,
  View
 } from "react-native"
import {
  Button,
  Input,
  Container,
  Item,
  Icon,
} from 'native-base'
import firebase from 'react-native-firebase'
import  styles  from './styles/signUpStyles'
import Text, { globalTextStyle } from 'Anyone/js/constants/Text'
import colors from './constants/colors'
import TouchableOpacity from './constants/TouchableOpacityLockable.js'
import Header from './header/header'
import { PulseIndicator } from 'react-native-indicators'


 class SignUp extends Component {

  static navigationOptions = {
    header: null,
  }

  constructor(props) {
    super(props)

    this.username = ""
    this.email = ""
    this.pass = ""
    this.confirm = ""
    this.state = {
      spinner: false
    }
  }

  componentDidMount() {

    firebase.auth().onAuthStateChanged(user => {
      if (user) {
        this.user = user
    // User is signed in.
  } else {
    // No user is signed in.
  }
})
  }

  render () {
    return (
    <Container style={styles.container}>
      <Header 
      hasBack={true}
      title={'Sign up'}
      />
      {this.state.spinner && <PulseIndicator color={colors.secondary}/>}
      <View style={{justifyContent: 'center', flex: 1}}>
      <Item style={styles.inputGrp}>
      <Icon name="person" style={{color: "#fff"}} />
        <Input
        placeholder="Username"
        onChangeText={u => this.username = u}
        placeholderTextColor="#FFF"
        style={styles.input}
        autoCapitalize={'none'}
        autoCorrect={false}
        keyboardType={'email-address'}
        />
        </Item>
      <Item style={styles.inputGrp}>
      <Icon name="mail" style={{color: "#fff"}} />
        <Input
        placeholder="Email"
        onChangeText={e => this.email = e}
        placeholderTextColor="#FFF"
        style={styles.input}
        autoCapitalize={'none'}
        autoCorrect={false}
        keyboardType={'email-address'}
        />
        </Item>
      <Item style={styles.inputGrp}>
      <Icon name="unlock" style={{color: "#fff"}} />
      <Input
        placeholder="Password"
        secureTextEntry={true}
        placeholderTextColor="#FFF"
        onChangeText={p => this.pass = p}
        style={styles.input}
        />
        </Item>
      <Item style={styles.inputGrp}>
      <Icon name="unlock" style={{color: "#fff"}} />
      <Input
        placeholder="Confirm Password"
        secureTextEntry={true}
        placeholderTextColor="#FFF"
        onChangeText={p => this.confirm = p}
        style={styles.input}
        />
        </Item>
        <TouchableOpacity 
        style={{backgroundColor: colors.secondary, padding: 10, paddingHorizontal: 20, alignSelf: 'center'}}
        onPress={() => {
          if (this.pass == this.confirm) {
            this.setState({spinner: true})
            if (this.username) {
              firebase.database().ref('/usernames/' + this.username).once('value')
              .then(snapshot => {
                if (snapshot.val()) {
                  Alert.alert('Sorry', 'That username is already in use')
                  this.setState({spinner: false})
                }
                else {
                  this.signup(this.email, this.pass)
                }
              }) 
            }
            else {
              Alert.alert('Sorry', 'Please choose a username')
              this.setState({spinner: false})
            }
          }
          else {
            Alert.alert('Please try again', 'Passwords do not match')
            this.setState({spinner: false})
          }
        }}
        >
        <Text style={{color: '#fff'}}>Sign up</Text>
        </TouchableOpacity>
        </View>
    </Container>
  )
  }

    async signup(email, pass) {

      try {
        await firebase.auth()
        .createUserWithEmailAndPassword(email, pass).then(({info, user}) => {
         let userData = {uid: user.uid, email: user.email, username: this.username}
         this.createUser(user.uid, userData, "")
         user.sendEmailVerification().then(()=> {
           this.props.goBack()
           Alert.alert('Account created', 'You must now verify your email using the link we sent you before you can login')
           this.setState({spinner: false})
         }).catch(error => {
          Alert.alert('Error', error.message)
        this.setState({spinner: false})
        })
       }).catch(error => {
        console.log(error)
        Alert.alert('Error', error.message)
        this.setState({spinner: false})
      })
      } catch (error) {
        console.log(error.toString())
        this.setState({spinner: false})
        Alert.alert(error.toString())
      }

}

createUser = (uid,userData,token) => {
  firebase.database().ref('admins').child(uid).once('value', snapshot => {
    const defaults = {
      uid,
      token,
      accountType: snapshot.val() ? 'admin' : 'standard',
    }
    //Alert.alert("Success", "Logged in as: " + userData.email)
    firebase.database().ref('users').child(uid).update({ ...userData, ...defaults })
    if (this.username) {
      firebase.database().ref('usernames').child(this.username).set(uid)
    }
})
  }

}

import { connect } from 'react-redux'
import { navigateBack } from './actions/navigation'
// const mapStateToProps = ({ home, settings, profile }) => ({
// })

 const mapDispatchToProps = dispatch => ({
   goBack: () => dispatch(navigateBack())
})

export default connect(null, mapDispatchToProps)(SignUp)
