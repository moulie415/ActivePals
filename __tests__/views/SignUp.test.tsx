import React from 'react';
import {shallow} from 'enzyme';
import SignUp from '../../app/views/SignUp';
import {SignUpNavigationProp} from '../../app/types/views/SignUp';

describe('SignUp', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(
        <SignUp navigation={{} as SignUpNavigationProp} />,
      );
      expect(component).toMatchSnapshot();
    });
  });
});
