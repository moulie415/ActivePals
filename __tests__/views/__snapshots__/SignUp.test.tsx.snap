// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
